src/lib.rs

```diff
@@ -44,36 +49,57 @@ mod server;
 /// Main Trait for an ABCI application. Provides generic responses for all callbacks
 /// Override desired callbacks as needed.  Tendermint makes 3 TCP connections to the
 /// application and does so in a synchonized manner.
-pub trait Application {
+pub trait Application: 'static + Send {
+    /// Signal the start of the server to the application, passing along an Exiter.
+    fn start(&mut self, _exiter: Exiter) {}
+
+    /// Signal the exit of the server to the application.
+    fn exit(&mut self) {}
+
     /// Query Connection: Called on startup from Tendermint.  The application should normally
     /// return the last know state so Tendermint can determine if it needs to replay blocks
     /// to the application.
-    fn info(&mut self, _req: &RequestInfo) -> ResponseInfo {
-        ResponseInfo::new()
+    fn info(&mut self, _req: RequestInfo, responder: Responder) {
+        let mut response = Response::new();
+        let res = ResponseInfo::new();
+        response.set_info(res);
+        let _ = responder.respond(response);
```

Comment:
A single call to `ResponseInfo::new()` is indeed simpler, and it's also semantically different so I don't think a comparison is in order. 

As to error handling, it might make sense for the application to send some errors to the protocol via an additional method on `Responder`. Alternatively, it could also use to `Exiter` to stop the protocol and server, if there is no way to recover from the error.

Reply:
I know the semantics are different, I was just pointing out how this change makes things a lot more verbose.
It might be better if the responder wrapped the specific response type in a `Response` before sending it through the channel, so we could do this:
```rust
 responder.respond_info(ResponseInfo::new())
```

Reply:
That could be another way to achieve it, and there you'd end-up with a lot more methods on the `Responder`. 

You could perhaps solve that problem by introducing a `AppResponse` enum, pass that to `respond`, which would match on it and internally create the response and call the appropriate `set_` method on it with the inner value of the enum variant.